misc/formatter/modules/formatter-core/src/main/java/org/ballerinalang/formatter/core/FormattingTreeModifier.java

```diff
@@ -278,7 +258,7 @@ public FunctionSignatureNode transform(FunctionSignatureNode functionSignatureNo
         Token openPara = getToken(functionSignatureNode.openParenToken());
         Token closePara = getToken(functionSignatureNode.closeParenToken());
 
-        NodeList<ParameterNode> parameters = this.modifyNodeList(functionSignatureNode.parameters());
+        SeparatedNodeList<ParameterNode> parameters = this.modifySeparatedNodeList(functionSignatureNode.parameters());
```

Comment:
Found a bug in TreeModifier.java. https://github.com/ballerina-platform/ballerina-lang/blob/c4f3e100fa1cbd4f003f689ecf90fe560d63689b/compiler/ballerina-parser/src/main/java/io/ballerinalang/compiler/syntax/tree/TreeModifier.java#L3100-L3101

* Here we have used `modifyGenericNodeList()` which only iterators through the alternative elements as we are parsing a `SeparatedNodeList` object, not  `NodeList`

* So when there's no modification that method simply returns the original SeparatedNodeList which is fine. 
But when there's a modification it return SeparatedNodeList which does not contain separators as no processing has been added with `getSeparator()`.

* Subsequently, when we are formatting when there's a modification, separators just get vanished. 

@SupunS shall I create an issue for the issue root?

Reply:
+1 please create an issue

Reply:
+1. Please create an issue